server/middlewares/csp.ts

```diff
@@ -0,0 +1,42 @@
+import * as helmet from 'helmet'
+
+const baseDirectives = {
+  defaultSrc: ["'none'"], // by default, not specifying default-src = '*'
+  connectSrc: ['*', 'data:'],
+  mediaSrc: ["'self'", 'blob:'],
+  fontSrc: ["'self'", 'data:'],
+  imgSrc: ["'self'", 'data:'],
+  scriptSrc: ["'self' 'unsafe-inline'"],
+  styleSrc: ["'self' 'unsafe-inline'"],
+  // objectSrc: ["'none'"], // only define to allow plugins, else let defaultSrc 'none' block it
+  formAction: ["'self'"],
+  frameAncestors: ["'none'"],
+  baseUri: ["'self'"],
+  pluginTypes: ["'none'"],
+  manifestSrc: ["'self'"],
+  frameSrc: ["'self'"], // instead of deprecated child-src / self because of test-embed
+  workerSrc: ["'self'"], // instead of deprecated child-src
+  upgradeInsecureRequests: true,
+  blockAllMixedContent: true
+}
+
+const baseCSP = helmet.contentSecurityPolicy({
+  directives: baseDirectives,
+  browserSniff: false,
+  reportOnly: true
+})
+
+const embedCSP = helmet.contentSecurityPolicy({
+  directives: Object.assign(baseDirectives, {
+    frameAncestors: ['*']
+  }),
+  browserSniff: false, // assumes a modern browser, but allows CDN in front
+  reportOnly: true
```

Comment:
With this configuration, violations will only be reported in the browser console. So it would be really hard to tell if it works, without asking all the users. It would be better if you either hardcode a report url, so admins can collect data with csp-logger, or if you included csp-logger which puts the reports in a known location.

Reply:
@Nutomic added in https://github.com/Chocobozzz/PeerTube/pull/1252/commits/1b92e937ff5aeac7366fa8004cf02681f17ed25c